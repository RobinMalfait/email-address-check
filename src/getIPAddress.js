const dns = require("dns");
import promisify from "./promisify";

const dnsLookup = promisify(dns.lookup);

async function getIPAddress(hostname) {
  const hostname_ip_regex = /(\[(.*)\])/;

  if (hostname_ip_regex.test(hostname)) {
    return hostname.replace("[", "").replace("]", "");
  }

  const [ip] = await dnsLookup(hostname);
  return ip;
}

export default getIPAddress;
