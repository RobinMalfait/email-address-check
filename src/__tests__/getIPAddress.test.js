import getIPAddress from "../getIPAddress";

describe("getIPAddress", () => {
  it("should return the ip address if the hostname already is an ip address", () => {
    return expect(getIPAddress("[127.0.0.1]")).resolves.toMatchSnapshot();
  });

  it("should return the ip address for the hostname", () => {
    return expect(getIPAddress("example.com")).resolves.toMatchSnapshot();
  });
});
