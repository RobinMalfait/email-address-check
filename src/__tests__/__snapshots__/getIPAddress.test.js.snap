// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getIPAddress should return the ip address for the hostname 1`] = `"93.184.216.34"`;

exports[`getIPAddress should return the ip address if the hostname already is an ip address 1`] = `"127.0.0.1"`;
